include/vcpkg/base/expected.h

```diff
@@ -184,6 +184,18 @@ namespace vcpkg
             return !value_is_error ? *m_t.get() : T(std::forward<Args>(or_args)...);
         }
 
+        template<class Func>
+        T value_or_lazy(Func&& func) &&
```

Comment:
I'm not convinced that this is so universally applicable that it should be here... even in this PR it seems questionable to report error text as a 'version'.